From 3fe21dde39d90eea1012a5b83182abb27809fd6c Mon Sep 17 00:00:00 2001
From: Mattias Busck <mattias.busck@hostmobility.com>
Date: Tue, 1 Nov 2022 15:48:25 +0100
Subject: [PATCH 10/11] HMX DT: Set wakeup-source on flexcan

wakeup-source seems necessary to wake up from tcan1146 when it pulls the
rxd pin low when entering standby from sleep.

Set pinctrl for nWKRQ_CANFD(wake cpu input).
---
 arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts b/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
index c55c4fdf858d..3de36d78f51e 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-var-dart-hmx1.dts
@@ -367,12 +367,14 @@ &flexcan1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_flexcan1>;
 	status = "okay";
+	wakeup-source;
 };
 
 &flexcan2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_flexcan2>;
 	status = "okay";
+	wakeup-source;
 };
 
 
@@ -752,6 +754,8 @@ MX8MP_IOMUXC_SAI5_RXD2__GPIO3_IO23				0x12
 			MX8MP_IOMUXC_SAI2_MCLK__GPIO4_IO27				0x1c0
 			MX8MP_IOMUXC_SAI2_TXD0__GPIO4_IO26				0x1c0
 
+			/* nWKRQ_CANFD */
+			MX8MP_IOMUXC_NAND_DQS__GPIO3_IO14				0x10
 			
 		>;
 	};
-- 
2.30.2

